### ISSUE TITLE
Not working with TeamCity v9.1.1

### ISSUE THREAD

**Issue by Darkreaper101** [parautenbach/TeamCity-HipChat-Notifier#58]
Good Day,

The HipChat notifier isn't working with the latest version of TeamCity v9.1.1
Currently the plugin installs perfectly, but link item to configure your HipChat settings isn't visible.

Do you maybe have a solution to this issue? :smile:

**Comment by mgoodfellow** [comment]
+1

Same issue here - The installed plugin can be seen in the Plugins list

![plugin](https://cloud.githubusercontent.com/assets/6501186/9272171/bac44342-4277-11e5-9146-74b42d06a179.PNG)

But no link for configuration!

**Comment by Darkreaper101** [comment]
@parautenbach Do you have any idea what could be causing this to happen?

**Comment by mgoodfellow** [comment]
I notice that it seems JetBrains were a bit harsh and broke plugin support in 9.1 - in their [plugin list](https://confluence.jetbrains.com/display/TW/TeamCity+Plugins) there are a few listed with 9.1 compatible forks:

Existing GitHub plugin: https://github.com/milgner/TeamCityGithub/
9.1 compatible fork: https://github.com/orybak/TeamCityGithub

Same for redmine.

I think it would have been useful in the TeamCity [release notes](https://confluence.jetbrains.com/display/TW/TeamCity+9.1+Release+Notes) or [what's new](https://confluence.jetbrains.com/display/TCD9/What%27s+New+in+TeamCity+9.1) to mention they were breaking support with existing plugins ;)

**Comment by batesbg** [comment]
I'm also having this issue.  Would be great to see a fix.

**Comment by jptoto** [comment]
Ditto that for 9.1.1.

**Comment by stpkys** [comment]
TeamCity 9.1.1, java version "1.8.0_45"
Error looks same as #57 

gins.spring.SpringPluginLoader - Failed to initialize spring context for plugin HipChat Notifier 
java.lang.ClassFormatError: Illegal local variable table length 6 in method com.whatsthatlight.teamcity.hipchat.HipChatConfiguration.setEvents(Lcom/whatsthatlight/teamcity/hipchat/HipChatEventConfiguration;)V

**Comment by andrejohansson** [comment]
+1 here

**Comment by RMSEddy** [comment]
Running 9.1.1 / 0.9.1 on Server 2012 R2 with both the TC-included JRE (1.8.0_45) and 1.8.0_60 Server JRE.

[2015-09-03 10:35:21,926]  ERROR - gins.spring.SpringPluginLoader - Failed to initialize spring context for plugin HipChat Notifier 
java.lang.ClassFormatError: Illegal local variable table length 48 in method com.whatsthatlight.teamcity.hipchat.HipChatConfiguration.<init>()V
	at java.lang.Class.getDeclaredConstructors0(Native Method)
	at java.lang.Class.privateGetDeclaredConstructors(Class.java:2671)
	at java.lang.Class.getDeclaredConstructors(Class.java:2020)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.determineCandidateConstructors(AutowiredAnnotationBeanPostProcessor.java:234)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.determineConstructorsFromBeanPostProcessors(AbstractAutowireCapableBeanFactory.java:1041)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1014)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:504)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:475)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:302)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:229)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:298)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:193)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:706)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:762)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:482)
	at jetbrains.buildServer.plugins.spring.SpringPluginLoader.pluginClassesLoaded(SpringPluginLoader.java:95)
	at sun.reflect.GeneratedMethodAccessor40.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at jetbrains.buildServer.util.EventDispatcher.dispatch(EventDispatcher.java:120)
	at jetbrains.buildServer.util.EventDispatcher$2.invoke(EventDispatcher.java:67)
	at com.sun.proxy.$Proxy17.pluginClassesLoaded(Unknown Source)
	at jetbrains.buildServer.plugins.PluginManagerImpl$2.visitPlugin(PluginManagerImpl.java:137)
	at jetbrains.buildServer.plugins.PluginsCollection.foreachLoadedPlugins(PluginsCollection.java:222)
	at jetbrains.buildServer.plugins.PluginManagerImpl.firePluginClassesLoaded(PluginManagerImpl.java:135)
	at jetbrains.buildServer.plugins.PluginManagerImpl.loadPlugins(PluginManagerImpl.java:81)
	at jetbrains.buildServer.web.plugins.PluginManagerConfigurator.initializePlugins(PluginManagerConfigurator.java:9)
	at jetbrains.buildServer.web.impl.BuildServerConfigurator.loadConfiguration(BuildServerConfigurator.java:27)
	at jetbrains.buildServer.maintenance.TeamCityDispatcherServlet$WebApplicationCreatorAndDestroyer.createApplication(TeamCityDispatcherServlet.java:14)
	at jetbrains.buildServer.maintenance.StartupProcessor.doApplicationStarting(StartupProcessor.java:60)
	at jetbrains.buildServer.maintenance.StartupProcessor.processConcreteStage(StartupProcessor.java:443)
	at jetbrains.buildServer.maintenance.StartupProcessor.processConcreteStageSafe(StartupProcessor.java:694)
	at jetbrains.buildServer.maintenance.StartupProcessor.processTeamCityLifecycle(StartupProcessor.java:202)
	at jetbrains.buildServer.maintenance.StartupProcessor.access$000(StartupProcessor.java:285)
	at jetbrains.buildServer.maintenance.StartupProcessor$1.run(StartupProcessor.java:2)
	at java.lang.Thread.run(Thread.java:745)

**Comment by jonathanrasquin** [comment]
+1 here

**Comment by scene316** [comment]
+1

**Comment by travislaynewilson** [comment]
+1

**Comment by ptgamr** [comment]
+1

**Comment by neronmoon** [comment]
+1

**Comment by stebet** [comment]
+1

**Comment by parautenbach** [comment]
I've upgraded my test instance to TC v9.1.1-37059 and the plugin v0.9.1 starts up fine, but trying to load the configuration page, I get the following, which indicates a change to the HipChat APIs (the page does load though):

```
2015-09-10 21:51:28,795 - ERROR - Could not get rooms 
org.codehaus.jackson.map.exc.UnrecognizedPropertyException: Unrecognized field "version" (Class com.whatsthatlight.teamcity.hipchat.HipChatRoom), not marked as ignorable
 at [Source: java.io.InputStreamReader@561871e4; line: 1, column: 248] (through reference chain: com.whatsthatlight.teamcity.hipchat.HipChatRooms["items"]->com.whatsthatlight.teamcity.hipchat.HipChatRoom["version"])
```

I need to know who is getting the same error (enable [debug logging](https://github.com/parautenbach/TeamCity-HipChat-Notifier#developers) if you haven't already). Those that have voted on this issue, could you please confirm exactly which error you're getting (i.e. a snippet of your log files please)?

**Comment by parautenbach** [comment]
I know it may sound audacious to ask this, but I have to check the most common things first. Did the problem for any of you started right after the upgrade? If so, there is a possibility that you didn't clear out the [TeamCity Home Directory](https://confluence.jetbrains.com/display/TCD9/TeamCity+Home+Directory) as per the [upgrade notes](https://confluence.jetbrains.com/display/TCD9/Upgrade). 

As some of the logs posted here indicate a configuration issue, could some of you paste your (safe) `hipchat.xml` config files (typically found in `.BuildServer/config/`)?

**Comment by parautenbach** [comment]
Unfortunately I can't yet reproduce this issue, so any additional information will be helpful. I did some reading, and did find [this insightful post](http://stackoverflow.com/questions/31566672/java-8-what-is-the-equivalent-of-usesplitverifier). Perhaps one of you could try the setting just to see if it has an effect, and it could then serve as a temporary workaround until I find the conflicting library. 

Other resources:
* [Java 7 Bytecode Verifier](http://chrononsystems.com/blog/java-7-design-flaw-leads-to-huge-backward-step-for-the-jvm)
* [How safe is it to use -XX:-UseSplitVerifier?](http://stackoverflow.com/questions/15253173/how-safe-is-it-to-use-xx-usesplitverifier)
* [Java 6 -> 7 migration ClassFormatError: Illegal local variable table length](http://stackoverflow.com/questions/21917985/java-6-7-migration-classformaterror-illegal-local-variable-table-length)

**Comment by jkordish** [comment]
@parautenbach As my usual upgrades I backup the directory and start fresh. I too am getting the error on the new 9.1.1 TC release and latest notifier plugin.

**Comment by stephen-czetty** [comment]
Added -XX:-UseSplitVerifier, now I get:

```
[2015-09-10 20:10:12,543]  ERROR - gins.spring.SpringPluginLoader - Failed to initialize spring context for plugin HipChat Notifier
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'hipChatConfiguration' defined in plugin: HipChat Notifier#HipChatNotifier.jar!/META-INF/build-server-plugin-hipchat-notifier.xml: Instantiation of bean failed; nested exception is java.lang.NoClassDefFoundError: com/vladium/emma/rt/RT
        at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:278)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1115)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1018)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:504)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:475)
        at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:302)
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:229)
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:298)
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:193)
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:706)
        at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:762)
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:482)
        at jetbrains.buildServer.plugins.spring.SpringPluginLoader.pluginClassesLoaded(SpringPluginLoader.java:95)
        at sun.reflect.GeneratedMethodAccessor56.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at jetbrains.buildServer.util.EventDispatcher.dispatch(EventDispatcher.java:120)
        at jetbrains.buildServer.util.EventDispatcher$2.invoke(EventDispatcher.java:67) 
        at com.sun.proxy.$Proxy16.pluginClassesLoaded(Unknown Source)
        at jetbrains.buildServer.plugins.PluginManagerImpl$2.visitPlugin(PluginManagerImpl.java:137)
        at jetbrains.buildServer.plugins.PluginsCollection.foreachLoadedPlugins(PluginsCollection.java:222)
        at jetbrains.buildServer.plugins.PluginManagerImpl.firePluginClassesLoaded(PluginManagerImpl.java:135)
        at jetbrains.buildServer.plugins.PluginManagerImpl.loadPlugins(PluginManagerImpl.java:81)
        at jetbrains.buildServer.web.plugins.PluginManagerConfigurator.initializePlugins(PluginManagerConfigurator.java:9)
        at jetbrains.buildServer.web.impl.BuildServerConfigurator.loadConfiguration(BuildServerConfigurator.java:27)
        at jetbrains.buildServer.maintenance.TeamCityDispatcherServlet$WebApplicationCreatorAndDestroyer.createApplication(TeamCityDispatcherServlet.java:14)
        at jetbrains.buildServer.maintenance.StartupProcessor.doApplicationStarting(StartupProcessor.java:60)
        at jetbrains.buildServer.maintenance.StartupProcessor.processConcreteStage(StartupProcessor.java:443)
        at jetbrains.buildServer.maintenance.StartupProcessor.processConcreteStageSafe(StartupProcessor.java:694)
        at jetbrains.buildServer.maintenance.StartupProcessor.processTeamCityLifecycle(StartupProcessor.java:202)
        at jetbrains.buildServer.maintenance.StartupProcessor.access$000(StartupProcessor.java:285)
        at jetbrains.buildServer.maintenance.StartupProcessor$1.run(StartupProcessor.java:2)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NoClassDefFoundError: com/vladium/emma/rt/RT
        at com.whatsthatlight.teamcity.hipchat.HipChatConfiguration.$VRi(HipChatConfiguration.java)
        at com.whatsthatlight.teamcity.hipchat.HipChatConfiguration.<clinit>(HipChatConfiguration.java)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
        at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:148)
        at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:125)
        at jetbrains.buildServer.spring.InstantiationStrategySelector$1.instantiate(InstantiationStrategySelector.java:77)
        at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:270)
        ... 32 more
Caused by: java.lang.ClassNotFoundException: Class 'com.vladium.emma.rt.RT' was not found
        at jetbrains.buildServer.plugins.classLoaders.TeamCityClassLoader.doLoadClass(TeamCityClassLoader.java:85)
        at jetbrains.buildServer.plugins.classLoaders.TeamCityClassLoader.loadClass(TeamCityClassLoader.java:40)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
        ... 42 more
```

I was able to get it to work by copying emma.jar into WEB-INF/lib, but perhaps this release accidently included the code coverage instrumentation?

**Comment by jkordish** [comment]
@stephen-czetty i can confirm that adding the emma.jar over fixes it. :)

**Comment by RMSEddy** [comment]
Workaround doesn't apply to Java 8. 
https://docs.oracle.com/javase/8/docs/technotes/tools/windows/java.html#BABCBGHF

**Comment by aalbertson** [comment]
Interestingly I'm having the same issue, but on 9.0.4. Installed plugin, link doesn't appear.

**Comment by jptoto** [comment]
Mine was a fresh install of TeamCity 9.1.1 on Windows 2012 R2. with the
included TC Java 1.8.0_45, JRE. Having the same issue.

On Fri, Sep 11, 2015 at 1:32 PM, aalbertson <notifications@github.com>
wrote:

> Interestingly I'm having the same issue, but on 9.0.4. Installed plugin,
> link doesn't appear.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/parautenbach/TeamCity-HipChat-Notifier/issues/58#issuecomment-139608044>
> .
>



-- 
JP Toto   |   james.p.toto@gmail.com   |   http://jptoto.jp   |   @jptoto
<http://twitter.com/jptoto>

**Comment by bhcleek** [comment]
I'm having this problem on TC 9.0.1 (seeing the same error as @RMSEddy). The only change to the server was installing this plugin.

**Comment by bhcleek** [comment]
Comparing this release with the previous release, the two zip files' sizes are significantly different. Digging into 0.9.1's zip file and unzipping the HipChatNotifier.jar reveals another HipChatNotifier.zip, which has yet another HipChatNotifier.jar. It looks like @stephen-czetty may be on to something - the build process for 0.9.1 was different than it was for 0.9.0...

**Comment by parautenbach** [comment]
Thank you everyone for helping to investigate! The prognosis is indeed correct. I am going to pull the 0.9.1 release and release a new version. My apologies to everyone that scratched their heads about this.

**Comment by parautenbach** [comment]
Fixed in [v0.9.2](https://github.com/parautenbach/TeamCity-HipChat-Notifier/releases/tag/0.9.2).
